internal/aws/awsutil/conn.go

```diff
@@ -37,14 +41,22 @@ import (
 
 type ConnAttr interface {
 	newAWSSession(logger *zap.Logger, roleArn string, region string, profile string, sharedCredentialsFile []string) (*session.Session, error)
-	getEC2Region(s *session.Session) (string, error)
+	getEC2Region(cfg awsSDKV2.Config) (string, error)
 }
 
 // Conn implements connAttr interface.
 type Conn struct{}
 
-func (c *Conn) getEC2Region(s *session.Session) (string, error) {
-	return ec2metadata.New(s).Region()
+func (c *Conn) getEC2Region(cfg awsSDKV2.Config) (string, error) {
+	clientIMDSV2Only, clientIMDSV1Fallback := CreateIMDSV2AndFallbackClient(cfg)
+	region, err := clientIMDSV2Only.GetRegion(context.Background(), &imds.GetRegionInput{})
+	if err != nil {
```

Comment:
Does it make sense to fallback for _all_ errors? Or would it be better to only fallback for specific errors? I am guessing there isn't just a single error that tells us it failed because imdvs2 is not enabled?